File: Vozcribe-Windows/Vozcribe/Services/SoftUpdateNudgePolicy.cs
using Vozcribe.Models;

namespace Vozcribe.Services;

public static class SoftUpdateNudgePolicy
{
    public static bool ShouldShow(
        VersionStatus status,
        bool hasCompletedOnboarding,
        string? lastShownVersion)
    {
        if (!status.IsUpdateAvailable) return false;
        if (!hasCompletedOnboarding) return false;
        if (string.IsNullOrEmpty(status.LatestVersion)) return false;
        if (lastShownVersion == status.LatestVersion) return false;
        return true;
    }
}

File: VoxType-Mac/VoxType/Utilities/SoftUpdateNudgePolicy.swift
import Foundation

enum SoftUpdateNudgePolicy {
    static func shouldShow(
        result: VoxTypeAPIClient.VersionCheckResult,
        hasCompletedOnboarding: Bool,
        lastShownVersion: String?
    ) -> Bool {
        guard case .updateAvailable = result.status else { return false }
        guard hasCompletedOnboarding else { return false }
        guard let latestVersion = result.latestVersion,
              !latestVersion.isEmpty else { return false }
        if lastShownVersion == latestVersion { return false }
        return true
    }
}
